utils/fetch-utils.js

```diff
@@ -0,0 +1,96 @@
+import { writeFile, readFile, mkdir } from 'fs/promises';
+import { fetch } from '@adobe/fetch';
+
+const delay = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+async function fetchIndex(url) {
+    console.log('Fetching entries and saving locally');
+    const index = await fetch(url);
```

Comment:
Does this need error handling?
```suggestion
    const index = await fetch(url);
    if (!index.ok) throw new Error(`Error fetching index: ${index.status}`);
```

Reply:
That's a good idea, thanks